doc/Development_Documentation/21_Deployment/03_Multi_Environment.md

```diff
@@ -33,16 +33,126 @@ var/config/system.php
+
+```
+app/config/pimcore/
+```
+
+As a starting point, you can inspire your new YAML config file from the existing YAML config files for dev and prod environment in `app/config/pimcore/` and `pimcore/lib/Pimcore/Bundle/CoreBundle/Resources/config/pimcore/`.
+
+For instance, the most minimal config, reproducting prod environment, will be:
+
+```
 imports:
     - { resource: config.yml }
+
+monolog:
+    handlers:
+        main:
+            type:         fingers_crossed
+            action_level: error
+            buffer_size: 2000
+            handler:      nested
+        nested:
+            type:  stream
+            path:  "%kernel.logs_dir%/%kernel.environment%.log"
+            level: debug
+        console:
+            type:  console
 ```
 
+A config file, reproducting dev environment (and using specific bundles, see below how to register them) will be like this:
+
+```
+imports:
+    - { resource: config.yml }
+
+framework:
+    router:
+        resource: "%kernel.root_dir%/config/routing_dev.yml"
+        strict_requirements: true
+    profiler: { only_exceptions: false }
+
+web_profiler:
+    toolbar: true
+    intercept_redirects: false
+
+monolog:
+    handlers:
+        main:
+            type: stream
+            path: "%kernel.logs_dir%/%kernel.environment%.log"
+            level: debug
+            channels: ["!event"]
+        console:
+            type:   console
+            channels: ["!event"]
+
+pimcore:
+    error_handling:
+        render_error_document: false
+
+    web_profiler:
+        toolbar:
+            excluded_routes: ~
+
+    translations:
+        # enables support for the pimcore_debug_translations magic parameter
+        # see https://pimcore.com/docs/5.0.x/Development_Tools_and_Details/Magic_Parameters.html
+        debugging:
+            enabled: true
+```
+
+If you use some specific bundles, you need to register them in `app\AppKernel.php`.
+For instance for an new environment called `staging` using web profiler, you should add in the method `registerBundlesToCollection`:
+```
+use Pimcore\Bundle\GeneratorBundle\PimcoreGeneratorBundle;
+use Sensio\Bundle\DistributionBundle\SensioDistributionBundle;
+use Sensio\Bundle\GeneratorBundle\SensioGeneratorBundle;
+use Symfony\Bundle\DebugBundle\DebugBundle;
+use Symfony\Bundle\WebProfilerBundle\WebProfilerBundle;
+...
+	public function registerBundlesToCollection(BundleCollection $collection)
+	{
+		//...
+		
+		// environment specific bundles
+		if (in_array($this->getEnvironment(), ['staging'])) {
+			$collection->addBundles([
+				new DebugBundle(),
+				new WebProfilerBundle(),
+				new SensioDistributionBundle()
+			], 80);
+			// add generator bundle only if installed
+			if (class_exists('Sensio\Bundle\GeneratorBundle\SensioGeneratorBundle')) {
+				$collection->addBundle(
+					new SensioGeneratorBundle(),
+					80, // priority
+					['dev'] // will use the dev config files of the bundle
```

Comment:
This means the bundle will only be loaded in the `dev` environment and should be changed to `staging` in your example (comment should be removed or changed).